packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
 import { Controller } from 'ember-runtime';
-import { run } from 'ember-metal';
-
-import { Application } from 'ember-application';
-import { Router } from 'ember-routing';
-import { compile } from 'ember-template-compiler';
-import {
-  Component,
-  setTemplates,
-  setTemplate
-} from 'ember-glimmer';
-import { jQuery } from 'ember-views';
-
-let App, appInstance;
-
-function prepare() {
-  setTemplate('components/expand-it', compile('<p>hello {{yield}}</p>'));
-  setTemplate('application', compile('Hello world {{#expand-it}}world{{/expand-it}}'));
-}
-
-function cleanup() {
-  run(() => {
-    try {
-      if (App) {
-        App.destroy();
-      }
-      App = appInstance = null;
-    } finally {
-      setTemplates({});
-    }
-  });
-}
-
-QUnit.module('Application Lifecycle - Component Registration', {
-  setup: prepare,
-  teardown: cleanup
-});
-
-function boot(callback, startURL = '/') {
-  run(() => {
-    App = Application.create({
-      name: 'App',
-      rootElement: '#qunit-fixture'
-    });
+import { Component } from 'ember-glimmer';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 
-    App.deferReadiness();
+moduleFor('Application Lifecycle - Component Registration', class extends ApplicationTestCase {
 
-    App.Router = Router.extend({
-      location: 'none'
+  ['@test The helper becomes the body of the component'](assert) {
+    this.addTemplate('application', 'Hello world {{#expand-it}}world{{/expand-it}}');
+    this.addComponent('expand-it', {
+      template: '<p>hello {{yield}}</p>'
```

Comment:
Ah I'm seeing some of these tests explicitly test registered component vs resolved items. I'll leave some notes.